modules/core/src/main/java/org/apache/ignite/cache/CacheMetrics.java

```diff
@@ -705,4 +707,13 @@
      * @return {@code true} when cache topology is valid for writing.
      */
     public boolean isValidForWriting();
+
+    /**
+     * Checks if there were any tx key collisions last time.
+     * Interval for check specified throught: {@link IgniteSystemProperties#IGNITE_DUMP_TX_COLLISIONS_INTERVAL} or
+     * {@link TransactionsMXBean#setTxKeyCollisionsInterval(int)}.
+     *
+     * @return Key collisions and appropriate queue size string representation.
+     */
+    public String getTxKeyCollisions();
```

Comment:
Please consider adding @NotNull, it's useful cause we use `getTxKeyCollisions()` as a parameter for `writeUTF()` that will throw NPE in case of null String.